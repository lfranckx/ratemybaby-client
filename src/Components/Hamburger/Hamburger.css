/* Hamburger Menu Design*/
.menu-wrap {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: 1px;
    left: 15rem;
    z-index: 1;
    min-width: 120pt;
    padding: 10px;
    margin: 0;
}

.toggle-label {
    display: flex;
    align-items: center;
    justify-content: space-around;
}

.menu-label {
    font-size: 18pt;
    color: rgb(255, 74, 74);
    font-weight: bold;
    visibility: hidden;
}

.menu-wrap .hamburger {
    cursor: pointer;
    width: 25px;
    height: 25px;
    display: flex;
    align-items: center;
    justify-content: center; 
    background-color: none;
}

/* Hamburger Line */
.hamburger-container {
    background-color: #fff;
    border: 2px solid rgb(255, 74, 74);
    padding: 10px;
    border-radius: 50px;
    z-index: 2px
}

.menu-wrap .hamburger > div {
    position: relative;
    width: 100%;
    height: 2px;
    background-color: rgb(255, 74, 74);
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.4s ease;
}

.menu-wrap .hamburger > div:before,
.menu-wrap .hamburger > div:after {
    content: '';
    position: absolute;
    top: -10px;
    width: 100%;
    height: 2px;
    background: inherit;
}
/* Moves lines down */
.menu-wrap .hamburger > div:after {
    top: 10px;
}
/* Toggler animate */
.menu-wrap .checked > div {
    transform: rotate(135deg)
}
/* Turn lines into X */
.menu-wrap .checked > div:before,
.menu-wrap .checked > div:after {
    top: 0;
    transform: rotate(90deg);
}
/* Rotate on hover when checked */
.menu-wrap .checked:hover > div {
    transform: rotate(225deg)
}

/* Menu, ul, and li Styling */
.menu-wrap .menu {
    width: 100%;
    height: 100%;
    visibility: hidden;
    text-align: left;
    padding: 5px;
}

.menu-wrap .menu > ul {
    padding: 0;
    margin: 0;
}

.menu-wrap .menu > ul > li {
    list-style: none;
    margin: .8rem .5rem 1.4rem;
    visibility: hidden;
}

/* Show/Hide Menu */
.menu-wrap .checked ~ .menu {
    visibility: visible;
}

.menu-wrap .checked > ul > li {
    visibility: visible
}

.menu-wrap .checked  > ul > li > a {
    visibility: visible;
}

.show-label {
    visibility: visible;
}

.active {
    background: #fff;
    border: solid 4px rgb(255, 74, 74);
    border-radius: 8px;
}

.disabled {
    background: none;
    border: none;
    height: 10px;
    width: 10px;
}

/* Toddler-Links Styling */
.toddler-links {
    background: white;
    color: rgb(255, 74, 74);
    border: solid 2px rgb(255, 74, 74);
    border-radius: 40px;
    padding: 8px 20px;
    border-radius: 5px;
    font-weight: bold;
    margin-top: 10px;
    visibility: hidden;
}

.toddler-links:hover {
    background: rgb(255, 74, 74);
    color: white;
}

/* MEDIA QUERIRES */
@media (max-width: 700px) {
    .menu-wrap {
        top: 3.5rem;
        left: 1rem;
        padding: 5px;
    }

    .menu-wrap .hamburger {
        width: 18px;
        height: 18px;
    }

    .menu-label {
        font-size:20px;
        margin-right: 6px;
    }

    .menu ul li {
        text-align: left;
    }

    .toddler-links {
        font-size: 14px;
        padding: 6px 15px;
    }
}