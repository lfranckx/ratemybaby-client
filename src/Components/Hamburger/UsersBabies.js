import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import BabyContext from '../../Contexts/BabyContext'
import './Hamburger.css'

export default class ThingListItem extends Component {

    static contextType = BabyContext

    renderUsersBabies() {
        const {baby} = this.props
        return (
            <li>
                <Link 
                    className="toddler-links"
                    to={`/profile/${baby.id}`}
                >
                    {baby.baby_name}
                </Link>
            </li>
        )
    }

    render() {
        const { error } = this.context
        
        return (  
            <> 
            {error
                ? <p className='red'>There was an error, try again</p>
                : this.renderUsersBabies()} 
            </>
        )        
    }              
}



        
        