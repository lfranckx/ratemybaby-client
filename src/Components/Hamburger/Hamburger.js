import React, { Component } from 'react'
import './Hamburger.css'
import BabyContext from '../../Contexts/BabyContext'
import BabyApiService from '../../Services/baby-api-service'
import UsersBabies from './UsersBabies'
import { min } from 'date-fns'

export default class Hamburger extends Component { 
    static defaultProps = {
        match: { params: {} }
    }

    static contextType = BabyContext

    state = { 
        isChecked: false
    }

    componentDidMount() {
        BabyApiService.getByParentId()
            .then(res => {
                this.context.setUsersBabies(res)
            })
            .catch(this.context.setError)
    }

    toggleCheck = () => {
        this.setState({ isChecked: !this.state.isChecked })
    }

    render() {
        const { usersBabies } = this.context
        console.log(this.state);
        
        if (usersBabies.length === 0) {
            return <></>
        }

        const menuWrap = document.getElementById('menu-wrap')
        if (this.state.isChecked) {
            menuWrap.classList.add('checked')
        } 
        if (this.state.isChecked === false) {
            menuWrap.classList.remove('checked')
            menuWrap.classList.add('notChecked')
        }
        

        return (
            <div id="menu-wrap">
                <label id="menu-label">Babies</label>
                <input 
                    name="babies"
                    type='checkbox' 
                    className='toggler'
                    onClick={() => {this.toggleCheck()}} />
                <div className="hamburger"><div></div></div>
                <div className="menu">
                    <ul>
                    {usersBabies.map(baby => (
                        <UsersBabies 
                            key={baby.id}
                            baby={baby} />
                        ))}
                    </ul>
                </div>
            </div>
        )
    }
}