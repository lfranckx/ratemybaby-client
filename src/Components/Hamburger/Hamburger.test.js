import React from 'react';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Hamburger from './Hamburger';

const nullBaby = {
    baby: null
}

const context = {
    active: false,
    usersBabies: [],
    babies: [],
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    toggleActive: () => {},
    setNotActive: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    setUsersBabies: () => {}
}

describe(`Hamburger component`, () => {
    
    it(`renders page without crashing`, () => {
        const wrapper = mount(<Hamburger />, { context: context })
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})