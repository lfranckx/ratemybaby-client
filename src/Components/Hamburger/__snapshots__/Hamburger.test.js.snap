// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger component renders page without crashing 1`] = `<Hamburger />`;
